src/app/pages/system/view-enclosure/components/enclosure-disks/enclosure-disks.component.ts

```diff
@@ -292,11 +286,11 @@ export class EnclosureDisksComponent implements AfterContentInit, OnDestroy {
     this.themeUtils = new ThemeUtils();
     this.diskTemperatureService.listenForTemperatureUpdates();
 
-    core.register({ observerClass: this, eventName: 'DiskTemperatures' }).pipe(untilDestroyed(this)).subscribe((evt: DiskTemperaturesEvent) => {
+    this.diskTemperatureService.temperature$.pipe(untilDestroyed(this)).subscribe((data) => {
```

Comment:
We shouldn't make the subject public. Instead, let's make the subject private and make an observable of the subject public.